### ISSUE TITLE
getMilStdSymbolasIcon Returning Incorrectly Colored Icon

### ISSUE THREAD

**Issue by david-melendez** [missioncommand/old-mil-sym-java#36]
Attempting to retrieve an icon for the following graphics returns an icon that has the wrong color for the fill of the icon.  Some of the icons has no fill color when they should and some have a slightly different color that the properly rendered graphic.

```
|METOC|OCEANIC|OCEANOGRAPHY|BIOLUMINESCENCE|VRD LEVEL 1-2
|METOC|OCEANIC|OCEANOGRAPHY|BIOLUMINESCENCE|VRD LEVEL 2-3
|METOC|OCEANIC|GEOPHYSICS/ACOUSTICS|MINE WARFARE BOTTOM DESCRIPTORS|MIW BOTTOM TYPE|C2
|METOC|OCEANIC|GEOPHYSICS/ACOUSTICS|MINE WARFARE BOTTOM DESCRIPTORS|MIW BOTTOM TYPE|C3
|METOC|OCEANIC|MAN-MADE STRUCTURES|SUBMERGED CRIB
|METOC|OCEANIC|HYDROGRAPHY|COASTAL HYDROGRAPHY|ISLAND
|METOC|OCEANIC|HYDROGRAPHY|COASTAL HYDROGRAPHY|BEACH
|METOC|OCEANIC|HYDROGRAPHY|COASTAL HYDROGRAPHY|FORESHORE|FORESHORE LINE
|METOC|OCEANIC|HYDROGRAPHY|PORTS AND HARBORS|FACILITIES|DRYDOCK
|METOC|OCEANIC|HYDROGRAPHY|PORTS AND HARBORS|FACILITIES|OFFSHORE LOADING FACILITY (AREA)
|METOC|OCEANIC|HYDROGRAPHY|AIDS TO NAVIGATION|PERCHES/STAKES|PERCHES/STAKES (AREA)
|METOC|OCEANIC|HYDROGRAPHY|DANGERS/HAZARDS|UNDERWATER DANGER/HAZARD
|METOC|OCEANIC|HYDROGRAPHY|DANGERS/HAZARDS|DISCOLORED WATER
|METOC|OCEANIC|OCEANOGRAPHY|BEACH SLOPE|STEEP
|METOC|OCEANIC|GEOPHYSICS/ACOUSTICS|MINE WARFARE BOTTOM DESCRIPTORS|MIW-BOTTOM SEDIMENTS|SOLID ROCK
```

**Comment by michael-spinelli** [comment]
Deutch, you're going to have to run through and check METOC area fill colors.  They're not matching 2525.

Island (WO-DHCI-----A--) for example, is pretty far off.  Drydock (WO-DHPMD----A--) is another.

I believe I fixed all the colors for the icon variants of these symbols.

**Comment by mdeutch3** [comment]
Fixed METOC fill colors have been checked in.

**Comment by michael-spinelli** [comment]
fixed in release v0.1.16

**Comment by david-melendez** [comment]
Based on v0.1.16, this issue is partially fixed.  The following symbol icon colors appear to be correct:

- VDR Level 1-2
- VDR Level 2-3
- Submerged Crib
- Island
- Beach
- Foreshore Line
- Perches/Stakes (Area)
- Solid Rock

The following have inconsistent information in the standard.  Each of these symbols have a blue fill according to the standard, but the color in the displayed graphic (as seen in the document standard) for _Submerged Crib_ is blue, but is cyan for _Underwater Danger/Hazard_ and _Discolored Water_.   All the icons are blue (RGB = 0000FF) in color, but the rendered KML color is 00BFFF for _Underwater Danger/Hazard_ and _Discolored Water_.

- Submerged Crib
- Underwater Danger/Hazard
- Discolored Water



The following are incorrect:

- C2 - icon appears black
- C3 - icon appears black
- Offshore Loading Facility (Area) - icon appears black
- Steep - icon appears black

**Comment by michael-spinelli** [comment]
Checked in an update.  For Underwater Danger/Hazard and Discolored Water I made them Cyan to match the example since no specific RGB values were called out in the 2525 for those symbols.

**Comment by mdeutch3** [comment]
Of the 3 multi-point symbols that were listed, the standard says blue. Does the standard say they are RGB 0000FF? We are going by what they display. For example we have symbols where it says brown but displays different colors. Where they do specify an RGB color we use the RGB value specified.

**Comment by david-melendez** [comment]
I am not sure if there is a specific place that specifies what RGB values are to be used for what METOC symbols.  I just noted that all three symbols say blue fill in the standard, but two of the symbols are rendered in Cyan.

 

I received the following email from Michael Spinelli regarding his actions for this issue:

 

Checked in an update. For Underwater Danger/Hazard and Discolored Water I made them Cyan to match the example since no specific RGB values were called out in the 2525 for those symbols.

 

 

David Melendez

 <mailto:david.melendez@nova-tech-solutions.com> david.melendez@nova-tech-solutions.com

Software Engineer

Nova Tech Solutions
